src/layout/msWorkspace/horizontalPanel.js

```diff
@@ -110,6 +112,9 @@ var HorizontalPanel = GObject.registerClass(
 
         vfunc_allocate(box, flags) {
             SetAllocation(this, box, flags);
+            if (!this.tilingIcon || this.iconSize != box.get_height()) {
+                this.buildIcon(box.get_height());
```

Comment:
Why do you want to do it in the allocate instead of the `panel-size-changed` ?

Reply:
This does not work if the button isn't visible: if the button/icon has not been displayed (rendered) yet, there are a couple of errors in the log about the Actor and the icon won't be displayed if you change to that workspace (tested on both XOrg and Wayland). To reproduce it is easy: start MS with a couple of workspaces, open Settings and change panel size, when you activate the other workspaces the layout icon is missing.